js/okex.js

```diff
@@ -1835,6 +1835,67 @@ module.exports = class okex extends Exchange {
         return this.parseOrder (order, market);
     }
 
+    async cancelOrders (ids, symbol = undefined, params = {}) { // TODO : the original endpoint signature differs, according to that you can skip individual symbol and assign ids in batch. At this moment, `params` is not being used too.
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' canelOrders() requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = [];
+        // for client-order-ids
+        const clientOrderId = this.safeValue2 (params, 'clOrdId', 'clientOrderId');
+        if (Array.isArray (clientOrderId)) {
+            for (let i = 0; i < clientOrderId.length; i++) {
+                request.push ({
+                    'instId': market['id'],
+                    'clOrdId': clientOrderId[i],
+                });
+            }
+        } else if (typeof clientOrderId === 'string') {
+            request.push ({
+                'instId': market['id'],
+                'clOrdId': clientOrderId,
+            });
+        }
+        // for ids (only if client-order-ids were not set)
+        if (clientOrderId === undefined) {
+            if (typeof ids === 'string') {
+                const orderIds = ids.split (',');
+                for (let i = 0; i < orderIds.length; i++) {
+                    request.push ({
+                        'instId': market['id'],
+                        'ordId': orderIds[i],
+                    });
+                }
+            } else {
+                for (let i = 0; i < ids.length; i++) {
+                    request.push ({
+                        'instId': market['id'],
+                        'ordId': ids[i],
+                    });
+                }
+            }
+        }
+        const response = await this.privatePostTradeCancelBatchOrders (request); // dont extend with params, otherwise ARRAY will be turned into OBJECT
```

Comment:
Let's change lines 1847-1878 (we should only set order ids if client order ids were not specified, no need to check that twice):

```JavaScript
        if (clientOrderId === undefined) {
            if (typeof ids === 'string') {
                const orderIds = ids.split (',');
                for (let i = 0; i < orderIds.length; i++) {
                    request.push ({
                        'instId': market['id'],
                        'ordId': orderIds[i],
                    });
                }
            } else {
                for (let i = 0; i < ids.length; i++) {
                    request.push ({
                        'instId': market['id'],
                        'ordId': ids[i],
                    });
                }
            }
        } else if (Array.isArray (clientOrderId)) {
            for (let i = 0; i < clientOrderId.length; i++) {
                request.push ({
                    'instId': market['id'],
                    'clOrdId': clientOrderId[i],
                });
            }
        } else if (typeof clientOrderId === 'string') {
            request.push ({
                'instId': market['id'],
                'clOrdId': clientOrderId,
            });
        }
```

Reply:
I've updated, please lmk if I correctly understood it.